python/grass/temporal/temporal_granularity.py

```diff
@@ -221,30 +297,32 @@ def compute_relative_time_granularity(maps):
 
     """
 
+    if not maps:
+        return None
+
+    get_time_tuple = get_time_tuple_function(maps)
+
     # The interval time must be scaled to days resolution
     granularity = None
     delta = []
     # First we compute the timedelta of the intervals
-    for map in maps:
-        start, end = map.get_temporal_extent_as_tuple()
+    for i in range(len(maps)):
+        start, end = get_time_tuple(maps[i])
         if (start == 0 or start) and end:
             t = abs(end - start)
             delta.append(int(t))
 
-    # Compute the timedelta of the gaps
-    for i in range(len(maps)):
+        # Compute the timedelta of the gaps
         if i < len(maps) - 1:
-            relation = maps[i + 1].temporal_relation(maps[i])
-            if relation == "after":
-                start1, end1 = maps[i].get_temporal_extent_as_tuple()
-                start2, end2 = maps[i + 1].get_temporal_extent_as_tuple()
+            start2, end2 = get_time_tuple(maps[i + 1])
+            if _is_after(start2, start, end):
                 # Gaps are between intervals, intervals and
                 # points, points and points
-                if end1 and start2:
-                    t = abs(end1 - start2)
+                if end and start2:
+                    t = abs(end - start2)
                     delta.append(int(t))
-                if not end1 and start2:
-                    t = abs(start1 - start2)
+                if not end and start2:
+                    t = abs(start - start2)
                     delta.append(int(t))
 
     delta.sort()
```

Comment:
Hmm.... ` {}` creates a dict, Is there a literal for creating an empty set? `set({})` seems less efficient than ` set()` ...

Reply:
https://stackoverflow.com/questions/36674083/why-is-it-possible-to-replace-sometimes-set-with

Ya seems like you're right. Well, since it's only for the initialization, let's not worry about it, it remains at the order of micro optimisation.

A side note to take into account when you check the performance when refactoring. Starting with Python 3.10, performance is quite different (better) than 3.9. They worked on a project "Faster Python project", while the 50% at each step wasn't held
https://github.com/markshannon/faster-cpython/blob/master/plan.md
https://github.com/faster-cpython/ideas
There was still big changes under the hood. https://docs.python.org/3.10/whatsnew/3.10.html#optimizations

https://docs.python.org/3.11/whatsnew/3.11.html#optimizations

https://docs.python.org/3.11/whatsnew/3.11.html#whatsnew311-faster-cpython

https://docs.python.org/3.12/whatsnew/3.12.html#optimizations


So, tuning for specific inefficiencies of already old releases that only have security updates for a little while remaining (3.8 is really near the end), doesn't bring as much.